core/src/oracle_state.rs

```diff
@@ -587,3 +516,86 @@ impl<'a> DatapointBoxesSource for DatapointStage<'a> {
         Ok(res)
     }
 }
+
+/// Register scans and save in scanIDs.json (if it doesn't already exist), and wait for rescan to complete
+pub fn register_and_save_scans() -> std::result::Result<(), Error> {
+    let config = &ORACLE_CONFIG;
+    let local_oracle_address = config.oracle_address.clone();
+
+    let oracle_pool_participant_token_id = config.token_ids.oracle_token_id.clone();
+
+    let refresh_box_scan_name = "Refresh Box Scan";
+
+    let datapoint_contract_address =
+        OracleContract::checked_load(&config.oracle_box_wrapper_inputs.contract_inputs)?
+            .ergo_tree();
+    // If scanIDs.json exists, skip registering scans & saving generated ids
+    if !Path::new("scanIDs.json").exists() {
```

Comment:
We're going to need to check if token ids, etc. are changed by the UpdatePool command and register new scans and trigger the wallet re-scan. It's out of the scope of this PR, so I made https://github.com/ergoplatform/oracle-core/issues/110